### PR TITLE
[wip_lapi] Refactor metabase

### PR DESCRIPTION
**Pull request by AlteredCoder**
Do we want to put the archive that contains the dashboard templates in the pkg or in a s3 bucket ?

### CODE REVIEW

File: pkg/metabase/metabase.go [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+
+	return success, nil
+}
+
+func (m *Metabase) DumpConfig(path string) error {
+	data, err := json.Marshal(m.Config)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(path, data, os.ModePerm)
```
Comment (buixor) [inline pkg/metabase/metabase.go:397]: beware of the default permissions of the file, it contains credentials

File: cmd/crowdsec-cli/dashboard.go [hunk 2]
```diff
@@ -150,12 +193,65 @@ cscli dashboard remove --force
...
+			mb, err := metabase.NewMetabase(metabaseConfigPath)
+			if err != nil {
+				log.Fatalf(err.Error())
+			}
+
+			if err := mb.Login(); err != nil {
+				log.Fatalf(err.Error())
+			}
+
+			if err := mb.Export(metabaseExportPath); err != nil {
```
Comment (buixor) [inline cmd/crowdsec-cli/dashboard.go:214]: maybe dump a small message here, about where the export is stored ?

File: cmd/crowdsec-cli/dashboard.go [hunk 3]
```diff
@@ -150,12 +193,65 @@ cscli dashboard remove --force
 	cmdDashRemove.Flags().BoolVarP(&force, "force", "f", false, "Force remove : stop the container if running and remove.")
 	cmdDashboard.AddCommand(cmdDashRemove)
 
+	var cmdDashExport = &cobra.Command{
+		Use:   "export",
+		Short: "export the metabase container.",
+		Long:  `export the metabase container using docker.`,
```
Comment (buixor) [inline cmd/crowdsec-cli/dashboard.go:199]: I think we want to say what the export is containing

File: cmd/crowdsec-cli/dashboard.go [hunk 4]
```diff
@@ -150,12 +193,65 @@ cscli dashboard remove --force
...
+
+	var cmdDashImport = &cobra.Command{
+		Use:   "import",
+		Short: "import the metabase container.",
+		Long:  `import the metabase container using docker.`,
+		Args:  cobra.ExactArgs(0),
+		Example: `
+cscli dashboard import
+cscli dashboard import --force
+ `,
```
Comment (buixor) [inline cmd/crowdsec-cli/dashboard.go:230]: is it missing params ?

File: pkg/metabase/database.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+	idFloat, ok := body["id"].(float64)
+	if !ok {
+		return fmt.Errorf("bad database id type: %v", body)
+	}
+	d.ID = int(idFloat)
+
+	return nil
+}
+
+/*func (d *Database) Backup(data map[string]interface{}) error {
```
Comment (buixor) [inline pkg/metabase/database.go:126]: dead code

File: pkg/metabase/test/receive/receive.txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+GET /api/dashboard/3
+
```
Comment (buixor) [inline pkg/metabase/test/receive/receive.txt:2]: remove test file

File: pkg/metabase/test/send/send.txt [hunk 7]
```diff
@@ -0,0 +1,25 @@
+POST /api/dataset
```
Comment (buixor) [inline pkg/metabase/test/send/send.txt:1]: same

**Review by buixor — CHANGES_REQUESTED** [review]
lack of unicity on error returns, some linters errors, but looks good so far :+1:

**Comment by AlteredCoder** [comment]
TODO: unify error message
TODO: download .tar from s3 bucket for setup

**Pull request closed without merging**